### PR TITLE
Convert to @types

### LINKED ISSUE #67: Update template to use @types
**Enhancement**

The package template needs to be updated to use `@types` as far as can be currently done.  This also requires a couple changes to `typings.json` as well.

Both `dojo-compose` and `dojo-widgets` have been converted so those changes can be used as the pattern for updating the template.

Refs: dojo/meta#62

### PR DESCRIPTION
**Pull request by kitsonk**
**Type:** bug / feature

The following has been addressed in the PR:
- [X] There is a related issue
- [X] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
- [ ] Unit or Functional tests are included in the PR

**Description:**

Converts the template to use `@types` for most type resolution.

Resolves #67

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,17 +23,15 @@
...
-    "dts-generator": "~1.7.0",
-    "glob": "^7.0.3",
     "grunt": "~1.0.1",
     "grunt-dojo2": ">=2.0.0-beta.16",
-    "grunt-text-replace": "^0.4.0",
-    "intern": "^3.2.3",
-    "istanbul": "^0.4.3",
-    "remap-istanbul": "^0.6.4",
-    "tslint": "^3.11.0",
+    "intern": "^3.3.1",
```
Comment (agubler) [inline package.json]: fix this to `~3.3.2` for now

Comment (kitsonk) [inline package.json]: Fixed

**Comment by dylans** [comment]
This needs an update for the latest Intern 3.4.x.

**Comment by dylans** [comment]
We'll redo this one as this is now too out of date.

**Pull request closed without merging**